strawberryfields/circuitspecs/x.py

```diff
@@ -0,0 +1,256 @@
+            correspond to a lower and upper bound (inclusive).
+
+    Keyword Args
+        variable_name (str): the variable name to use when printing
+            the range
+    """
+
+    def __init__(self, *args, variable_name="x"):
+        self.ranges = [Range(*a, variable_name=variable_name) for a in args]
+
+    def __contains__(self, item):
+        for range_ in self.ranges:
+            if item in range_:
+                return True
+
+        return False
+
+    def __repr__(self):
+        return ", ".join([str(i) for i in self.ranges])
+
+
+class XSpecs(CircuitSpecs):
+    """Circuit specifications for the X class of circuits."""
+
+    short_name = "X"
+    modes = None
+    remote = True
+    local = True
+    interactive = False
+    allowed_sq_ranges = Ranges([0], [1.0], variable_name="r")
+
+    primitives = {
+        "S2gate",
+        "Sgate",
+        "MeasureFock",
+        "Rgate",
+        "BSgate",
+        "MZgate",
+        "Interferometer",
+    }
+
+    decompositions = {
+        "BipartiteGraphEmbed": {"mesh": "rectangular_symmetric", "drop_identity": False},
+    }
+
+    def compile(self, seq, registers):
+        # the number of modes in the provided program
+        n_modes = len(registers)
+
+        # Number of modes must be even
+        if n_modes % 2 != 0:
+            raise CircuitError("The X series only supports programs with an even number of modes.")
+
+        # Call the GBS compiler to do basic measurement validation.
+        # The GBS compiler also merges multiple measurement commands
+        # into a single MeasureFock command at the end of the circuit.
+        seq = GBSSpecs().compile(seq, registers)
+
+        # ensure that all modes are measured
+        if len(seq[-1].reg) != n_modes:
+            raise CircuitError("All modes must be measured.")
+
+        # Use the GaussianUnitary compiler to compute the symplectic
+        # matrix representing the Gaussian operations.
+        # Note that the Gaussian unitary compiler does not accept measurements,
+        # so we append the measurement separately.
+        meas_seq = [seq[-1]]
+        seq = GaussianUnitary().compile(seq[:-1], registers) + meas_seq
+
+        # determine the modes that are acted on by the symplectic transformation
+        used_modes = [x.ind for x in seq[0].reg]
+
+        # extract the compiled symplectic matrix
+        S = seq[0].op.p[0]
+
+        if len(used_modes) != n_modes:
+            # The symplectic transformation acts on a subset of
+            # the programs registers. We must expand the symplectic
+            # matrix to one that acts on all registers.
+            # simply extract the computed symplectic matrix
+            S = expand(seq[0].op.p[0], used_modes, n_modes)
+
+        half_n_modes = n_modes // 2
+
+        # Construct the covariance matrix of the state.
+        # Note that hbar is a global variable that is set by the user
+        cov = (sf.hbar / 2) * S @ S.T
```

Comment:
@nquesada worth double checking this: I wasn't sure if this should be the user-set hbar value or not.

Reply:
You can do any `hbar`, since that `hbar` is local and it is explicitly passe to `Amat`